python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -335,16 +345,15 @@ pyAltLaunchKernelBase(const std::string &name, MlirModule module,
   kernelReg();
 
   if (launch) {
-    auto &platform = cudaq::get_platform();
     auto uReturnOffset = static_cast<std::uint64_t>(returnOffset);
+    auto args = runtimeArgs.getArgs();
+    if (platform.get_remote_capabilities().isRemoteSimulator)
+      args.insert(args.begin(), reinterpret_cast<void *>(&mod));
+
     if (platform.is_remote() || platform.is_emulated()) {
-      auto *wrapper = new cudaq::ArgWrapper{mod, names, rawArgs};
-      auto dynamicResult = cudaq::altLaunchKernel(
-          name.c_str(), thunk, reinterpret_cast<void *>(wrapper), size,
-          uReturnOffset);
+      auto dynamicResult = cudaq::streamlinedLaunchKernel(name.c_str(), args);
```

Comment:
@amccaskey @schweitzpgi here is where I no longer pass the module to PyRemoteRESTPU.cpp